src/mpid/ch3/include/mpidpre.h

```diff
@@ -106,7 +106,7 @@ typedef int32_t MPIDI_Rank_t;
 typedef struct MPIDI_Message_match_parts {
     int32_t tag;
     MPIDI_Rank_t rank;
-    MPIR_Context_id_t context_id;
+    int context_id;
```

Comment:
This is really down in the weeds but the match optimization on the "whole" 64-bit value (see above comment) is gone since `MPIDI_Message_match_parts_t` would be 80 bits.

Reply:
Good catch!

We could add `MPIDI_Context_id_t` in the same way `MPIDI_Rank_t` is defined (which is locally in this file). They need be defined together anyway in order to realize the 64-bit optimization anyway.